metloom/pointdata/files.py

```diff
@@ -0,0 +1,253 @@
+"""
+Location to keep readers that pull in a flat file like a csv.
+"""
+import os
+
+import pandas as pd
+
+from metloom.pointdata import PointData
+from metloom.variables import VariableBase, SensorDescription
+from enum import Enum
+from pathlib import Path
+from datetime import datetime, timezone, timedelta
+import logging
+import requests
+from typing import List
+import geopandas as gpd
+import numpy as np
+
+
+LOG = logging.getLogger(__name__)
+
+
+class StationInfo(Enum):
+    """
+    Since there is not enough info via an API, csv readers rely on a list of stations
+    and info associated with them to build the calls. This should allow for a common
+    dataset with multiple stations to be verified and isolated
+    """
+
+    # Defined: Name, id, lat, long, http sub path to file
+
+    @property
+    def station_name(self):
+        return self.value[0]
+
+    @property
+    def station_id(self):
+        return self.value[1]
+
+    @property
+    def latitude(self):
+        return self.value[2]
+
+    @property
+    def longitude(self):
+        return self.value[3]
+
+    @property
+    def elevation(self):
+        return self.value[4]
+
+    @property
+    def path(self):
+        return Path(self.value[5])
+
+    @classmethod
+    def all_station_ids(cls):
+        return [e.station_id for e in cls]
+
+    @classmethod
+    def all_points(cls):
+        return [e.point for e in cls]
+
+    @classmethod
+    def from_station_id(cls, station_id):
+        result = None
+        for e in cls:
+            if station_id == e.station_id:
+                result = e
+                break
+        return result
+
+    @property
+    def point(self):
+        return gpd.points_from_xy([self.longitude],
+                                  [self.latitude],
+                                  [self.elevation])[0]
+
+
+class CSVPointData(PointData):
+    """
+    Some met station data is stored off in flat csv files. This class enables the
+    management of downloading those files while still allowing a similar interface
+    """
+    ALLOWED_VARIABLES = VariableBase
```

Comment:
Nitpicky, but you don't need `ALLOWED_VARIABLES = VariableBase` here since it is inheriting from PointData

Reply:
Nit pick away! I wouldnt PR it if I didnt want it to be the best lol, FIXED!